keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -129,12 +145,101 @@ def __init__(self):
+        images.  By default returns either a `tf.RaggedTensorSpec` matching the input
+        image spec, or a `tf.TensorSpec` matching the input image spec.
+        """
+        if self.force_output_dense_images:
+            return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+        if self.force_output_ragged_images or isinstance(images, tf.RaggedTensor):
+            ragged_spec = tf.RaggedTensorSpec(
+                shape=images.shape[1:],
+                ragged_rank=1,
+                dtype=self.compute_dtype,
+            )
+            return ragged_spec
+        return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+
+    # TODO(lukewood): promote to user facing API if needed
+    def _compute_bounding_box_signature(self, bounding_boxes):
+        ragged_spec = tf.RaggedTensorSpec(
+            shape=[None, bounding_boxes.shape[2]],
+            ragged_rank=1,
+            dtype=self.compute_dtype,
+        )
+        return ragged_spec
+
+    # TODO(lukewood): promote to user facing API if needed
+    def _compute_keypoints_signature(self, keypoints):
+        if isinstance(keypoints, tf.RaggedTensor):
+            ragged_spec = tf.RaggedTensorSpec(
+                shape=keypoints.shape[1:],
+                ragged_rank=1,
+                dtype=self.compute_dtype,
+            )
+            return ragged_spec
+
+        return tf.TensorSpec(
+            shape=keypoints.shape[1:],
+            dtype=self.compute_dtype,
+        )
+
+    # TODO(lukewood): promote to user facing API if needed
+    def _compute_target_signature(self, targets):
+        return tf.TensorSpec(targets.shape[1:], self.compute_dtype)
+
+    def _compute_output_signature(self, inputs):
+        fn_output_signature = {IMAGES: self.compute_image_signature(inputs[IMAGES])}
+        bounding_boxes = inputs.get(BOUNDING_BOXES, None)
+
+        if bounding_boxes is not None:
+            fn_output_signature[BOUNDING_BOXES] = self._compute_bounding_box_signature(
+                bounding_boxes
+            )
+
+        segmentation_masks = inputs.get(SEGMENTATION_MASKS, None)
+        if segmentation_masks is not None:
+            fn_output_signature[SEGMENTATION_MASKS] = self.compute_image_signature(
+                segmentation_masks
+            )
+
+        keypoints = inputs.get(KEYPOINTS, None)
+        if keypoints is not None:
+            fn_output_signature[KEYPOINTS] = self._compute_keypoints_signature(
+                keypoints
+            )
+
+        labels = inputs.get(LABELS, None)
+        if labels is not None:
+            fn_output_signature[LABELS] = self._compute_target_signature(labels)
+
+        return fn_output_signature
+
+    @staticmethod
+    def _any_ragged(inputs):
+        if isinstance(inputs[IMAGES], tf.RaggedTensor):
+            return True
+        if BOUNDING_BOXES in inputs:
+            return True
+        # TODO(lukewood): exhaustively check cases
```

Comment:
do we have to check for keypoints as well?

Reply:
Done